### ISSUE TITLE
Bus Guideways disappear at z12 and less

### ISSUE THREAD

**Issue by SomeoneElseOSM** [gravitystorm/openstreetmap-carto#2965]
As noted elsewhere, there aren't too many of these in OSM at the moment, but it would make sense to show them at z11 and z12.  Here's Cambridge at z12 now:

http://www.openstreetmap.org/#map=12/52.2625/0.0359

The guided busway to St Ives is as big a feature in the landscape as (say) the railway line out through Cherry Hinton, it would make sense to render it.  Currently there's no rendering below z13:

https://github.com/gravitystorm/openstreetmap-carto/blob/master/roads.mss#L2582

but adding this to cover z11 and z12 would be a big improvement:

    [zoom >= 11][zoom < 13] {
      line-width: 0.6;
      line-color: #6666ff;
      [zoom >= 12] { line-width: 1; }
    }

as per

https://github.com/SomeoneElseOSM/openstreetmap-carto-AJT/blob/master/roads.mss#L3094

This doesn't fix the other problems with guideways (due I suspect to it being in its own layer): https://github.com/gravitystorm/openstreetmap-carto/issues/2489 https://github.com/gravitystorm/openstreetmap-carto/issues/2897 , but it's a start.

**Comment by matthijsmelissen** [comment]
In general that rendering is quite awful.

**Comment by kocio-pl** [comment]
I don't see what's wrong with using railway scheme in the bus-related color?

**Comment by matthijsmelissen** [comment]
The bus guideway is now the most dominant feature on z13.

**Comment by SomeoneElseOSM** [comment]
> In general that rendering is quite awful.

Hehe - why don't you say what you really think :)

> The bus guideway is now the most dominant feature on z13.

Yes - the problem seems to be that the "guideways" layer used the railway line characteristics at z13 but with another colour, and did not take into account how much more "in your face" #6666ff looks when compared to the darkish grey used for railway=rail.

> I don't see what's wrong with using railway scheme in the bus-related color?

FWIW that's what I've done in the style that I look after - send it through as "railway=bus_guideway" (mainly to make the selection in the .mml a bit easier to understand) and use the existing railway code to handle bridges and tunnels etc.  I used #6699ff as a "less in your face colour" (though that may not work with the different water colours that OSM Carto uses) and, as above, used a significantly narrower line at zooms < 13 so that the Cambridge bus guideway is "about as obvious at those zooms as an unclassified road".

**Comment by kocio-pl** [comment]
No matter how we want to render it on z13+, it should be easy to make it being visible earlier as a thin, dark line. I even prefer to do it separately, since an easy fix shouldn't be blocked by discussed feature.

Could you prepare rendering test and PR for z<13? That would help a lot.

**Comment by Klaus-Tockloth** [comment]
Render bus_guideway similar to tram.

**Comment by SomeoneElseOSM** [comment]
With the rendering and colour change as above, example at z11

![camb_guideway_11_1](https://user-images.githubusercontent.com/5311740/33524823-940b598e-d81b-11e7-95fb-c61fc1bd02d9.png)

At z12:

![camb_guideway_12_1](https://user-images.githubusercontent.com/5311740/33524826-9d56be8e-d81b-11e7-8d4f-565cbe7cba8b.png)

and also at z13 to note the effect of the colour change:

![camb_guideway_13_1](https://user-images.githubusercontent.com/5311740/33524837-d0c9a880-d81b-11e7-85d8-52103136564b.png)

**Comment by SomeoneElseOSM** [comment]
@Klaus-Tockloth I don't think tram is an option, because the way that they're used within OSM is very different.  Trams run along city streets, bus guideways are used where there are no streets.  See for example here:

![camb_guideway_18_1](https://user-images.githubusercontent.com/5311740/33524882-77899540-d81c-11e7-9113-2ff655317ed9.png)

In OSM for comparison that's at http://www.openstreetmap.org/#map=18/52.29177/0.05476 .  Zoom out to see the lower zoom equivalents of the other screenshots.

**Comment by Klaus-Tockloth** [comment]
A bus_guideway is, seen from the user perspective, closer to a tram as to a railway.

**Comment by donaldrnoble** [comment]
I saw this linked in the WeeklyOSM.eu blog, and while the issue is closed, I have a comment

The blue colour used is VERY similar to a river IMO, especially for the lower zooms where it is rendered as a solid line. The pecked railway style at z13+ differentiates it OK.

It also seems that the icon for bus stops is a darker blue (It comes up as #0094d5, vs #679bf9 for me). I accept not wanting to make bus guideways overly prominent, but would it be possible to change the colour to a darker/less saturated blue to make them more distinct from waterways? 

Sorry I've not linked to the code directly, I get a bit confused by GitHub :)

**Comment by matkoniecz** [comment]
> The blue colour used is VERY similar to a river IMO, especially for the lower zooms where it is rendered as a solid line

As it is a new issue - can you create a new issue (direct link: https://github.com/gravitystorm/openstreetmap-carto/issues/new )

> Sorry I've not linked to the code directly, I get a bit confused by GitHub :)

Not a problem, this is not required. But on creating issue - can you give example of place where rendering is problematic?

**Comment by SomeoneElseOSM** [comment]
@donaldrnoble for info, I used http://www.openstreetmap.org/#map=11/52.2932/-0.0388 as an area to compare bus guideway colour with water and and cycleway.  If you compare the guideway to the lakes SE of St Ives and the Great Ouse I think it's just about different enough.  Too blue and at some zooms it looked to me like a cycleway.

The main reason for changing the colour though was to balance the visual effect of the guideway and the railway at http://www.openstreetmap.org/#map=12/52.2325/0.1378 (also z13).  With the original blue, the guideway was much more prominent.  To make it less prominent, you'd need to make it narrower (there's probably a technical term for this but not having any artistic abilities I've no idea what it is).  You can see the effect by looking [here](http://map.atownsend.org.uk/maps/map/map.html#zoom=16&lat=-24.99133&lon=135.16105) (that's the legend for a different map style).  The widths of the 3 colours of steps have been adjusted so that they have about the same visual weight - the black one is much narrower than the blue one, which in turn is narrower than the red one.

If someone's going to do that then to be honest it's worth handling guideways as a type of railway so that bridges and tunnels work properly too (there are already issues logged for those).

**Comment by sladen** [comment]
`tunnnel=yes`; `covered=yes` and layering under bridges appear to be broken/not rendered.

**Comment by kocio-pl** [comment]
Please open new ticket for this, with example places and screenshots so it's easier to check.

**Comment by SomeoneElseOSM** [comment]
Existing issues with bus guideways (including some examples) are logged in https://github.com/gravitystorm/openstreetmap-carto/issues/3359 , https://github.com/gravitystorm/openstreetmap-carto/issues/2897 and https://github.com/gravitystorm/openstreetmap-carto/issues/2489 .  I solved it in another style by treating guideways as a type of railway (via lua).  That sort of thing tends not to be done here so you'd probably need to duplicate railway tunnel handling into bus guideways.
